internal/route/routing_strategy.go

```diff
@@ -61,5 +75,98 @@ func (s *PriorityRoundRobinStrategy) RouteNextRequest(
 		s.logger.Debug("Did not find any healthy nodes in priority.", zap.Int("priority", priority))
 	}
 
+	// No healthy upstreams are available. If `alwaysRoute` is true, find an unhealthy upstream to route to anyway.
+	// TODO(polsar): At this time, the only unhealthy upstreams that can end up here are those due to high latency
+	//  or error rate. Pass ALL configured upstreams in `upstreamsByPriority`. Their health status should be indicated
+	//  in the UpstreamConfig.HealthStatus field.
+	if s.alwaysRoute {
+		s.logger.Warn("No healthy upstreams found but `alwaysRoute` is set to true.")
+
+		// If available, return an upstream that's unhealthy due to high latency rate.
+		if upstreamsByPriorityLatencyUnhealthy, ok := statusToUpstreamsByPriority[conf.ReasonLatencyTooHighRate]; ok {
+			upstream := getHighestPriorityUpstream(upstreamsByPriorityLatencyUnhealthy)
+			s.logger.Info(
+				"Routing to an upstream with high latency.",
+				zap.String("ID", upstream.ID),
+				zap.String("GroupID", upstream.GroupID),
+				zap.String("HTTPURL", upstream.HTTPURL),
+				zap.String("WSURL", upstream.WSURL),
+			)
+			// TODO(polsar): The call can return nil, but it shouldn't be possible. Should we still check?
```

Comment:
In my opinion, if something shouldn't be possible then we should fail hard and assert that it's not possible. Typically, some log warning is issued which gets buried in the noise and then you never know if what you thought was impossible is actually impossible. Impossible things that happen should be investigated deeper and our assumptions need to be re-evaluated.